### PR TITLE
Fix 28964 - Add configuration for localization pack

### LINKED ISSUE #28964: Define where PrestaShop fetch localization packs
## User stories

User story : localization pack
Description
As a developer I want an easy way to configure API , URL access to localization pack
In order to host myself those packs

## Business rules
1- Provide a configuration file to store URL / API call for the localisation pack
2- Provide placeholders in the value of the key for the URL / API ( such as lang , version (?) ... )
'https://i18n.prestashop-project.org/translations/%version%/%locale%/%locale%.zip'; =>placeholder => %version% , %locale%
3- - This configuration file should be human readable ( or at least 1 of the store option , see above )

### Acceptance test 1
Modify the url / API call for the localization path into the configuration file
When I install a localization pack, the installer should call this API / URL

### Acceptance test 2
Modify the url / API call for the localization path into the configuration file
When I upgrade a localization pack, the upgrade should call this API / URL

### Acceptance test 3
Modify the url / API call for the localization path into the configuration file
I the upgrade goes wrong , raise an exception

## Related 

Spike : https://github.com/PrestaShop/PrestaShop/issues/30409

### PR DESCRIPTION
**Pull request by FabienPapet**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | See issue.
| Type?             | new feature 
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | See issue.
| Fixed ticket?     | Fixes #28964
| Related PRs.     | https://github.com/PrestaShop/autoupgrade/pull/555

image

### CODE REVIEW

File: src/PrestaShopBundle/Form/Admin/Improve/International/Localization/AdvancedConfigurationType.php [hunk 1]
```diff
@@ -61,6 +61,13 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                     'The ISO 3166-1 alpha-2 identifier for the country/region where your web server is located, in lowercase (us, gb, fr, sp, ru, pl, nl, etc.).',
                     'Admin.International.Help'
                 ),
-            ]);
+            ])
+            ->add('translation_pack_url', TextType::class, [
+                'label' => $this->trans(
+                    'Translation pack url',
+                    'Admin.International.Feature'
+                ),
```
Comment (kpodemski) [inline src/PrestaShopBundle/Form/Admin/Improve/International/Localization/AdvancedConfigurationType.php:69]: I suggest adding a help text showing available variables `%version%`, `%locale%`

File: install-dev/data/xml/configuration.xml [hunk 2]
```diff
@@ -892,5 +892,8 @@ Country</value>
     <configuration id="PS_PRODUCT_REDIRECTION_DEFAULT" name="PS_PRODUCT_REDIRECTION_DEFAULT">
       <value>404</value>
     </configuration>
+    <configuration id="PS_LANGUAGE_I18N_URL" name="PS_LANGUAGE_I18N_URL">
+      <value>jpg</value>
```
Comment (kpodemski) [inline install-dev/data/xml/configuration.xml:902]: 👋🏻

File: src/PrestaShopBundle/Form/Admin/Improve/International/Localization/AdvancedConfigurationType.php (hunk 1 above)
Comment (FabienPapet) [reply src/PrestaShopBundle/Form/Admin/Improve/International/Localization/AdvancedConfigurationType.php:69]: ping @l-delin @fabienvallon for wording on the help text.

File: classes/Language.php [hunk 3]
```diff
@@ -1235,7 +1235,12 @@ public static function downloadLanguagePack($iso, $version, &$errors = [])
     public static function downloadXLFLanguagePack($locale, &$errors = [], $type = self::PACK_TYPE_SYMFONY)
     {
         $file = self::getPathToCachedTranslationPack($locale, $type);
-        $url = (self::PACK_TYPE_EMAILS === $type) ? self::EMAILS_LANGUAGE_PACK_URL : self::SF_LANGUAGE_PACK_URL;
+
+        $baseUrl = self::EMAILS_LANGUAGE_PACK_URL;
```
Comment (Progi1984) [inline classes/Language.php:1239]: No way to modify this URL ?

File: src/PrestaShopBundle/Form/Admin/Improve/International/Localization/AdvancedConfigurationType.php (hunk 1 above)
Comment (l-delin) [reply src/PrestaShopBundle/Form/Admin/Improve/International/Localization/AdvancedConfigurationType.php:69]: Hello! Two things:

1. By "translation pack", do you mean "localization pack"? 
2. Why do we need a help text showing variables (%version%, %locale%) when it seems they're already shown in the URL? see > 
image

I'm not sure a help text is needed here

File: classes/Language.php (hunk 3 above)
Comment (FabienPapet) [reply classes/Language.php:1239]: It can be done, but it is outside of the scope of this PR.

File: classes/Language.php [hunk 4]
```diff
@@ -1235,7 +1235,12 @@ public static function downloadLanguagePack($iso, $version, &$errors = [])
...
+        if ($type === self::PACK_TYPE_SYMFONY) {
+            $baseUrl = Configuration::get('PS_LANGUAGE_I18N_URL', null) ?: self::SF_LANGUAGE_PACK_URL;
```
Comment (kpodemski) [inline classes/Language.php:1241]:
```suggestion
            $baseUrl = Configuration::get('PS_LANGUAGE_I18N_URL') ?: self::SF_LANGUAGE_PACK_URL;
```

`null` is by default

File: classes/Language.php (hunk 3 above)
Comment (Progi1984) [reply classes/Language.php:1239]: Really ?!

**Review by zuk3975 — DISMISSED** [review]

**Review by kpodemski — CHANGES_REQUESTED** [review]
Hi @FabienPapet 

There are some conflicts.

File: src/PrestaShopBundle/Form/Admin/Improve/International/Localization/AdvancedConfigurationType.php (hunk 1 above)
Comment (FabienPapet) [reply src/PrestaShopBundle/Form/Admin/Improve/International/Localization/AdvancedConfigurationType.php:69]: Updated `Translation pack url` to `Localization pack url`

**Review by matks — DISMISSED** [review]

**Comment by matks** [comment]
Sanity tests failing but I am not sure it's related to your PR

**Comment by kpodemski** [comment]
@FabienPapet remember about PR for the `autoupgrade` module 👍🏻

**Comment by FabienPapet** [comment]
Done !

File: src/PrestaShopBundle/Form/Admin/Improve/International/Localization/AdvancedConfigurationType.php [hunk 5]
```diff
@@ -61,6 +61,13 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                     'The ISO 3166-1 alpha-2 identifier for the country/region where your web server is located, in lowercase (us, gb, fr, sp, ru, pl, nl, etc.).',
                     'Admin.International.Help'
                 ),
-            ]);
+            ])
+            ->add('translation_pack_url', TextType::class, [
+                'label' => $this->trans(
+                    'Localisation pack url',
```
Comment (l-delin) [inline src/PrestaShopBundle/Form/Admin/Improve/International/Localization/AdvancedConfigurationType.php]:
```suggestion
                    'Localization pack URL',
```

**Review by jolelievre — CHANGES_REQUESTED** [review]
Thanks @FabienPapet 

but actually I don't think this should be configured in database. The initial plan for this configuration was to improve our application configuration, this is a parameter that should be configured on the environment via a configuration file, and env variable or some other ways to define. But not via the DB configuration which should be left for parameters editable in the BO only.

This work on improving the application configuration ahs been left out for a while, we didn't have time to handle it in 8.0 We didn't invest any time for it in 8.1 because it's not in the product's plans. This is the perfect example of technical feature that never get prioritized.

**Comment by FabienPapet** [comment]
ping @fabienvallon

**Pull request closed without merging**